tsnake/grid.py

```diff
@@ -358,34 +310,36 @@ def _compute_intersection(self, snake: TSnake) -> [Point]:
                     #     edge, node1.position[0, 0], node1.position[0,1], node2.position[0, 0], node2.position[0, 1],
                     #     index, intersect_pt
                     # ))
-
         return intersections
 
-    def get_snake_intersections(self) -> [[Point]]:
+    def get_snake_intersections(self, snakes: TSnakes) -> [[Point]]:
         """
         Compute intersections between all snakes on the grid and the grid.
-        \nArgs:\n
-        * None
-        \nReturn:\n
+        Args:
+        * List of T-Snakes
+        Return:
         * list(list(Point)) containing the intersection points for each snake
         """
-        intersections = []
-        for snake in self._snakes:
-            intersections.append(self._compute_intersection(snake))
-        return intersections
-
+        return [self._compute_intersection(snake) for snake in snakes]
 
-### TODOS ###
-# 1. snake updates - Joe
-# 2. gan mask -> snake -> gan mask - Cole
-# 3. algo phase 1: grid intersections - Allen
-# 4. algo phase 2: turning nodes on / off, remove inactive points - Eric
+    def reparameterize(self, snakes: TSnakes):
+        '''
+        Takes a list of T-Snake instances and returns a new list of TSnakes
+        (which have possibly been split/merged/etc).
+        If no snakes are split/merged, then:
+        1) the snakes are presented in the same order as before, AND
+        2) each snake's nodes are presented in the same order.
+        '''
+        new_snakes = []
+        for snake in snakes:
+            # Compute intersections in counter-clockwise direction,
+            # so subsequent normal calculation is performed correctly.
+            # Create new snake nodes in same order as intersections were computed.
+            # Initialize snake in same order as intersections.
+            new_snakes.append(snake)
```

Comment:
Looks like I was tired when I wrote the comment last night, but obviously we don't just want to return the same snakes. Reparametrize should call a couple other functions for each snake, such as compute_intersections(), and an initialize_snake().